src/windows/wslaservice/inc/wslaservice.idl

```diff
@@ -102,12 +109,102 @@ struct _VIRTUAL_MACHINE_SETTINGS {
 } VIRTUAL_MACHINE_SETTINGS;
 
 
-typedef
-struct _WSLA_SESSION_SETTINGS {
+
+struct WSLA_SESSION_SETTINGS {
     LPCWSTR DisplayName;
-    // Details TBD.
-} WSLA_SESSION_SETTINGS;
+    LPCWSTR StoragePath;
+
+    // TODO: Termination callback, flags
+};
+
 
+struct WSLA_REGISTRY_AUTHENTICATION_INFORMATION
+{
+    int Dummy; // Dummy value for the .idl file to compile.
+    // TODO: Details TBD.
+};
+
+
+struct WSLA_IMAGE_INFORMATION
+{
+    LPWSTR Name;
+    LPWSTR Hash;
+    ULONGLONG Size;
+    ULONGLONG DownloadTimestamp;
+};
+
+struct WSLA_PROCESS_OPTIONS
+{
+    LPCSTR Executable;
+    [unique] LPCSTR CurrentDirectory;
+    [size_is(CommandLineCount)] LPCSTR* CommandLine;
+    ULONG CommandLineCount;
+    [unique, size_is(EnvironmentCount)] LPCSTR* Environment;
+    ULONG EnvironmentCount;
+    [unique, size_is(EnvironmentCount)] WSLA_PROCESS_FD *Fds;
+    int FdsCount;
+};
+
+struct WSLA_VOLUME
+{
+    LPCSTR HostPath;
+    LPCSTR ContainerPathPath;
+};
+
+struct WSLA_PORT_MAPPING
+{
+    USHORT HostPort;
+    USHORT ContainerPort;
+};
+
+struct WSLA_CONTAINER_OPTIONS
+{
+    LPCWSTR Image;
+    LPCWSTR Name;
+    struct WSLA_PROCESS_OPTIONS* InitProcessOptions;
+    [unique, size_is(VolumesCount)] struct WSLA_VOLUME* Volumes;
+    ULONG VolumesCount;
+    [unique, size_is(PortsCount)] struct WSLA_PORT_MAPPING* Ports;
+    ULONG PortsCount;
+    DWORD Flags; // GPU, Privileged, ...
+    // TODO: List specific GPU devices.
+    // TODO: Flags on wether the caller wants to override entrypoint, args, or both.
+    ULONGLONG ShmSize;
+};
+
+enum WSLA_CONTAINER_STATE
+{
+    WslaContainerStateInvalid = 0,
+    WslaContainerStateCreated = 1,
+    WslaContainerStateRunning = 2,
+    WslaContainerStateExited = 3,
+    WslaContainerStateFailed = 4,
+    // TODO: More states might be added to reflect all nerdctl's states.
+};
+
+enum WSLA_PROCESS_STATE
+{
+    WslaProcessStateUnknown = 0,
+    WslaProcessStateRunning = 1,
+    WslaProcessStateExited = 2,
+    WslaProcessStateSignalled = 3
+};
+
+[
+    uuid(1AD163CD-393D-4B33-83A2-8A3F3F23E608),
+    pointer_default(unique),
+    object
+]
+interface IWSLAProcess : IUnknown
```

Comment:
I did some more thinking and experimentation, and it turns out that having a process has multiple benefits: 

- It will easily allow us to have multiple people watching the same process
- It will allow us to have multiple implementation of that class in the future, potentially with a different IO relay logic
- It solves the "zombie" issue since now we'll now when callers drop the last reference to a given process